mkosi/kmod.py

```diff
@@ -55,6 +55,39 @@ def filter_kernel_modules(
     return sorted(modules)
 
 
+def filter_firmware(
+    root: Path,
+    firmware: set[Path],
+    *,
+    include: Iterable[str],
+    exclude: Iterable[str],
+) -> set[Path]:
+    if not include and not exclude:
+        return firmware
+
+    if exclude:
+        remove = set()
```

Comment:
This isn't required, you can just remove from `firmware` directly instead of gathering in a separate set first

Reply:
You cannot remove elements from `firmware` while you're iterating it... see below

Reply:
Ahah, indeed, nvm